DynamicProgramming/EditDistance.java

```diff
@@ -45,11 +48,12 @@ public static int minDistance(String word1, String word2) {
                 //if last two chars equal
                 if (c1 == c2) {
                     //update dp value for +1 length
-                    dp[i + 1][j + 1] = dp[i][j];
+                    /* This Line gives array.access.unsafe.high warning because the compiler is unable to statically verify that i+1 is less than length of the array "dp"*/
```

Comment:
In general, these kinds of comments shouldn't just say what the error is: they should also give an explanation of why the error is wrong (or right).

Reply:
So in this case, the comment should explain how we know that `i + 1` is less than the length of `dp`

Reply:
I have improved the comments.